library/alloc/src/string.rs

```diff
@@ -2224,6 +2224,62 @@ impl ToString for char {
     }
 }
 
+#[stable(feature = "u8_to_string_specialization", since = "1.999.0")]
+impl ToString for u8 {
+    #[inline]
+    fn to_string(&self) -> String {
+        let mut result = String::with_capacity(3);
+        let mut n = *self;
+        if n >= 100 {
+            result.push((b'0' + n / 100) as char);
+            n %= 100;
+        }
+        if !result.is_empty() || n >= 10 {
+            result.push((b'0' + n / 10) as char);
+            n %= 10;
+        };
+        result.push((b'0' + n) as char);
+        result
+    }
+}
+
+// 2 digit decimal look up table
+static DEC_DIGITS_LUT: &[u8; 200] = b"0001020304050607080910111213141516171819\
+      2021222324252627282930313233343536373839\
+      4041424344454647484950515253545556575859\
+      6061626364656667686970717273747576777879\
+      8081828384858687888990919293949596979899";
+
+#[stable(feature = "i8_to_string_specialization", since = "1.999.0")]
+impl ToString for i8 {
+    #[inline]
+    fn to_string(&self) -> String {
+        let mut n = *self;
+        let mut vec: Vec<u8> = if n < 0 {
```

Comment:
How's the generated code if you operate on `String` and `char` instead of `Vec` and `u8`? That would get rid of the unsafe.

Reply:
It's a fair few additional instructions using only safe rust.